### PR TITLE
Move search implementation to q-server

### PR DESCRIPTION
**Pull request by manuelroth**
- This PR moves the search implementation to the q-server
- There is now an endpoint which takes requests in the following form:
```http
GET /search?
tool=imageslider&
createdBy=manuel.roth@nzz.ch&
department=Storytelling%20/%20Infografik&
publication=nzz&
searchString=Bundesrat&
limit=18
```

### CODE REVIEW

File: client/src/resources/ItemStore.js [hunk 1]
```diff
@@ -133,68 +133,40 @@ export default class ItemStore {
...
-        queries.push(
-          `(tool:"${filter.selected}" OR tool:${filter.selected.replace(
-            new RegExp("_", "g"),
-            "-"
-          )})`
-        );
-
-        // we do have a specific tool filter, so we do not want to have the onlyTools processed
-        onlyTools = null;
+        searchUrl = searchUrl + "tool=" + filter.selected + "&";
```
Comment (benib) [inline client/src/resources/ItemStore.js]: Please use https://nodejs.org/api/querystring.html, construct an object of all query parameters and then `querystring.stringify` in the end.

Comment (manuelroth) [reply client/src/resources/ItemStore.js]: Good idea. I have implemented this using the URLSearchParams web api as this is on client side -> https://github.com/nzzdev/Q-editor/pull/97/commits/8f15c0039e09d0f58bda3a31a47b69fc2b4f7d30

File: client/src/resources/ItemStore.js [hunk 2]
```diff
@@ -133,113 +133,82 @@ export default class ItemStore {
...
-      this.items[row.doc._id];
+      item.addConf(doc);
+      this.items[doc._id];
       return item;
     });
-
     return {
       items: items,
-      total_rows: data.total_rows,
+      moreItemsAvailable: data.docs.length === limit,
```
Comment (benib) [inline client/src/resources/ItemStore.js]: please add a comment explaining this for our future selfs :-)

Comment (manuelroth) [reply client/src/resources/ItemStore.js]: Yes sure -> https://github.com/nzzdev/Q-editor/pull/97/commits/571147d29afdc35a077c49cce1355e3a2299385f

**Comment by benib** [comment]
This will be in the next release.

**Pull request merged**